lib/timescaledb/migration_helpers.rb

```diff
@@ -99,6 +99,8 @@ def create_hypertable(table_name,
     #   SQL
     #
     def create_continuous_aggregate(table_name, query, **options)
+      raise ArgumentError, 'query must be a string' unless query.is_a?(String)
```

Comment:
Just thinking about old migrations that are already in place.

```suggestion
      query = query.to_sql if query.respond_to(:to_sql)
      raise ArgumentError, 'query must be a string or implement the `to_sql` method ' unless query.is_a?(String)
```

Reply:
You're right to think about backwards compatibility, but if I apply your changes then this fixes nothing.

The problem is caused by allowing people to pass an ActiveRecord::Relation as an input to the method.

If you want to preserve backwards compatibility for now, we could look into making it a warning instead of an exception.